### PR TITLE
Add releasing for mac

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/PaddlePaddle/Paddle/issues/11900

### CODE REVIEW

File: doc/fluid/dev/releasing_process_en.md [hunk 1]
```diff
@@ -50,6 +50,33 @@ pop-up box, choose the current release branch and click "Run Build" button. You
...
+(10.11, 10.12, 10.13) and different python installs (python.org, homebrew, etc.),
+you must build the package ***exactly*** following below steps:
+
+Build steps:
+
+1. install python from python.org downloads, and make sure it's currently in use
+   in your system.
+1. `export MACOSX_DEPLOYMENT_TARGET=10.11`, use `10.11` is enough for recent versions.
+1. `git clone https://github.com/PaddlePaddle/Paddle.git && cd Paddle && mkdir build && cd build`
+1. `cmake -DWITH_GPU=OFF -DWITH_MKL=OFF -DWITH_SYSTEM_BLAS=OFF  ..`, make sure the output of `cmake` command is using the correct python interpreter installed from python.org
```
Comment (gangliao) [inline doc/fluid/dev/releasing_process_en.md:66]: We  can emphasize the CMake's Python args, for instance,

```bash 
-DPYTHON_EXECUTABLE=(which python)
 
-DPYTHON_LIBRARY=/usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7.dylib
```

Comment (typhoonzero) [reply doc/fluid/dev/releasing_process_en.md:66]: The reason is described at #11900, run `python -c "import distutils.util; print(distutils.util.get_platform())"` can get the python supported platform. brew installed python often is the same to your current MacOS's version.

Comment (typhoonzero) [reply doc/fluid/dev/releasing_process_en.md:66]: Install python from python.org, the platform is `macosx-10.6-intel` which can be wide.

**Review by luotao1 — APPROVED** [review]
LGTM！

**Pull request merged**